shared_memory_client.py

```diff
@@ -0,0 +1,170 @@
+from pickle import loads, dumps
+from tcp_communication import TCP_Connection
+from messages import Request, Response
+import SharedMemoryWrapper
+
+
+class SharedMemoryClient:
+    BUFFER_SIZE = 1024
+    ERROR = 0
+    SMT_INIT = 1
+    SMT_VERSION = 2
+    SMT_CREATE_TOPIC = 3
+    SMT_PUBLISH = 4
+    SMT_GET_BY_COUNTER = 5
+    SMT_GET_LATEST = 6
+    SMT_GET_OLDEST = 7
+    SMT_GET_PUBLISH_COUNT = 8
+    SMT_CLEAR_HISTORY = 9
+    EXIT = 999
+
+    def __init__(self, server_port, server_ip="127.0.0.1"):
+        self._connection = TCP_Connection(server_port, True, server_ip)
```

Comment:
It's better to receive the connection as an argument in the constructor (constructor should receive an interface (e.g. IComm_Device instead of concrete class).
What will happen if in a year from now we will want to change the communication protocol to UDP?

As a rule of thumb - the creation of an object in a constructor is problematic and should be avoided.

Reply:
good idea, thank you!
**done!**